benchmark/agbenchmark/app.py

```diff
@@ -1,78 +1,60 @@
 import datetime
-import uuid
-from collections import defaultdict, deque
-from pathlib import Path
-
-import httpx
-
-from agbenchmark.agent_protocol_client import (
-    AgentApi,
-    ApiClient,
-    ApiException,
-    Configuration,
-)
-from agbenchmark.reports.processing.report_types_v2 import BenchmarkRun
-from agbenchmark.schema import TaskEvalRequestBody
-from agbenchmark.utils.utils import write_pretty_json
-
-configuration = Configuration(host="http://localhost:8000" + "/ap/v1")
-
+import glob
 import json
+import logging
 import os
 import sys
+import uuid
+from collections import defaultdict, deque
+from pathlib import Path
 from typing import Any, Optional
 
+import httpx
 import psutil
-from fastapi import APIRouter, FastAPI
-from fastapi import (
-    HTTPException as FastAPIHTTPException,  # Import HTTPException from FastAPI
-)
-from fastapi import Request, Response
+from agent_protocol_client import AgentApi, ApiClient, ApiException, Configuration
+from agent_protocol_client.models import Task, TaskRequestBody
+from fastapi import APIRouter, FastAPI, HTTPException, Request, Response
 from fastapi.middleware.cors import CORSMiddleware
-
-from agbenchmark.execute_sub_process import execute_subprocess
-from agbenchmark.schema import Task, TaskRequestBody
-
-sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
-from fastapi import FastAPI
 from pydantic import BaseModel, Extra
 
-router = APIRouter()
-import glob
-
-# Change the current working directory to the benchmark path
-# home_path = find_absolute_benchmark_path()
-# os.chdir(home_path)
-
-general_command = ["poetry", "run", "agbenchmark", "start", "--backend"]
+from agbenchmark.config import AgentBenchmarkConfig
+from agbenchmark.execute_sub_process import execute_subprocess
+from agbenchmark.reports.processing.report_types_v2 import (
+    BenchmarkRun,
+    Metrics,
+    RepositoryInfo,
+    RunDetails,
+    TaskInfo,
+)
+from agbenchmark.schema import TaskEvalRequestBody
+from agbenchmark.utils.data_types import ChallengeData
+from agbenchmark.utils.utils import write_pretty_json
 
-import psutil
+sys.path.append(str(Path(__file__).parent.parent))
 
-challenges_path = os.path.join(os.path.dirname(__file__), "challenges")
+logger = logging.getLogger(__name__)
 
-json_files = deque(
+CHALLENGES: dict[str, ChallengeData] = {}
```

Comment:
Why the CAPS here

Reply:
Because it's a global that is used as a constant datasource at runtime.